datafusion/core/src/datasource/listing/url.rs

```diff
@@ -424,6 +433,13 @@ mod tests {
         let b = ListingTableUrl::parse("../bar/./foo/../baz").unwrap();
         assert_eq!(a, b);
         assert!(a.prefix.as_ref().ends_with("bar/baz"));
+
+        let url = ListingTableUrl::parse("../foo/*.parquet").unwrap();
```

Comment:
I am probably missing something here, but how does this test the new code? I don't see it passing in `ignore_subdirectory`

Reply:
Yes, it's really not obvious here (actually in `ListingTableUrl.contains`), I would create a sqllogictest as you suggested